GPU/Common/VertexDecoderCommon.cpp

```diff
@@ -1326,8 +1326,23 @@ void VertexDecoder::DecodeVerts(u8 *decodedptr, const void *verts, int indexLowe
 	int count = indexUpperBound - indexLowerBound + 1;
 	int stride = decFmt.stride;
 	if (jitted_) {
+		// Check alignment before running the decoder, as we may crash if it's bad (as should the real PSP but doesn't always)
+		bool bad = false;
+		if (biggest == 4) {
+			if (((uintptr_t)verts & 3) != 0)
+				bad = true;
+		} else if (biggest == 2) {
+			if (((uintptr_t)verts & 1) != 0)
+				bad = true;
+		}
+
 		// We've compiled the steps into optimized machine code, so just jump!
-		jitted_(ptr_, decoded_, count);
+		if (!bad) {
+			jitted_(ptr_, decoded_, count);
+		} else {
+			// Not really sure what to do here... zero the verts to be safe?
```

Comment:
No, but we should. Though most cases of this is probably games going haywire (for example through corrupting framebuffer reads) and trying to execute garbage data as a display list, as I don't believe the PSP supports misaligned vertex data.